hazelcast/src/main/java/com/hazelcast/client/impl/spi/impl/ClientClusterServiceImpl.java

```diff
@@ -89,14 +95,31 @@ private MemberListSnapshot(int version, LinkedHashMap<UUID, Member> members, UUI
         }
     }
 
-    public ClientClusterServiceImpl(ILogger logger) {
+    public ClientClusterServiceImpl(LifecycleService lifecycleService, ILogger logger) {
+        this.lifecycleService = lifecycleService;
+        this.lifecycleListenerID = this.lifecycleService.addLifecycleListener(event -> {
+            if (event.getState() == LifecycleEvent.LifecycleState.CLIENT_CONNECTED) {
+                clientConnected = true;
+            } else if (event.getState() == LifecycleEvent.LifecycleState.CLIENT_DISCONNECTED
+                    || event.getState() == LifecycleEvent.LifecycleState.SHUTDOWN
+                    || event.getState() == LifecycleEvent.LifecycleState.SHUTTING_DOWN) {
+                clientConnected = false;
+            }
+        });
         this.logger = logger;
     }
 
     public Cluster getCluster() {
         return new ClientClusterProxy(this);
     }
 
+    public boolean isEnterprise() {
+        if (this.clientConnected) {
```

Comment:
```suggestion
        if (!this.clientConnected) {
```
And your test fails at the moment.